packages/fetchai/connections/soef/connection.py

```diff
@@ -258,6 +259,7 @@ def __init__(
 
         self._token_storage_path = token_storage_path
         if self._token_storage_path is not None:
+            self._token_storage_path = os.path.abspath(self._token_storage_path)
```

Comment:
@DavidMinarsch 250fffd

Reply:
Not ideal, the problem is that we don't want to change the content of the packages. Config object should contain the agent dir so we can use that to build the right path?

Reply:
> we don't want to change the content of the packages

Not sure I follow this point. At the moment that line is executed, the local directory is the AEA project directory, and not the package directory.
In other words, if I have the following:
```
aea_project/
    ...
    vendor/
        path/to/soef/
            ...
            connection.yaml
```

the value of `token_storage_path` is appended to `aea_project`, and not `aea_project/path/to/soef`. This is true also for other files, like `libp2p_node.log`. So this change only stores the absolute path to avoid ambiguity in other parts of the SOEF connection code (as it happens for aea launch)

Reply:
> Config object should contain the agent dir so we can use that to build the right path?
This might be useful for many other purposes, we should open an issue for that.